### PR TITLE
Detect when cancel is clicked on web

### PR DESCRIPTION
**Pull request by letranloc**

### CODE REVIEW

**Review by ttpho — APPROVED** [review]
LGTM 💯 

@letranloc Thank you for the awesome PR.

@miguelpruivo @GeertJohan 
Please review merge this PR. (bow)

File: lib/src/file_picker_web.dart [hunk 1]
```diff
@@ -108,15 +108,34 @@ class FilePickerWeb extends FilePicker {
       });
     }
 
+    void cancelledEventListener(_) {
+      window.removeEventListener('focus', cancelledEventListener);
+
+      // This listener is called before the input changed event,
+      // and the `uploadInput.files` value is still null
+      // Wait for results from js to dart
+      Future.delayed(Duration(milliseconds: 500)).then((value) {
```
Comment (miguelpruivo) [inline lib/src/file_picker_web.dart:117]: Can we use a microtask here instead? (`scheduleMicrotask(callback)`)?

**Review by miguelpruivo — COMMENTED** [review]
I see what you did here. It's sort of a workaround I'd say, but it should work (in theory, I didn't have the chance to test it out yet).

Please, check the comments above.

File: lib/src/file_picker_web.dart (hunk 1 above)
Comment (letranloc) [reply lib/src/file_picker_web.dart:117]: Thank you

Tested with `scheduleMicrotask(callback)` and it doesn't work: the `callback` is called too soon

**Comment by miguelpruivo** [comment]
Thank you.

**Pull request merged**